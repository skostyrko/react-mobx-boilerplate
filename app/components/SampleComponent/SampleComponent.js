import React from 'react'

import styles from './SampleComponent.scss'
console.log(styles)
export default function Button (props) {
  return <button className={styles.bordered} onClick={props.onClick}>{props.children}</button>
}
