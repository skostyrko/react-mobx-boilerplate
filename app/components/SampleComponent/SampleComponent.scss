@import '../../styles/_variables.scss';

.bordered {
  border: red;
}
