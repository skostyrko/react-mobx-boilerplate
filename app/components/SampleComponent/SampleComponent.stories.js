import React from 'react'
import { storiesOf, action } from '@kadira/storybook'
import { withKnobs, text } from '@kadira/storybook-addon-knobs'
import { WithNotes } from '@kadira/storybook-addon-notes'

import SampleComponent from './SampleComponent'

const stories = storiesOf('Button', module)
stories.addDecorator(withKnobs)

stories.add('with text (with knob)', () => (
  <SampleComponent onClick={action('clicked')}>{text('Label', 'Hello Button')}</SampleComponent>
))
.add('with some emoji (with notes)', () => (
  <WithNotes notes={'Some info on component'}>
    <SampleComponent onClick={action('clicked')}>😀 😎 👍 💯</SampleComponent>
  </WithNotes>
))
