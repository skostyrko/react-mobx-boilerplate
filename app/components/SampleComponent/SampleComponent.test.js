import React from 'react'
import SampleComponent from './SampleComponent'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <SampleComponent>Test</SampleComponent>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
