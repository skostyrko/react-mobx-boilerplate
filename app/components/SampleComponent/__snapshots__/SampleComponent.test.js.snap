// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="bordered"
  onClick={undefined}
>
  Test
</button>
`;
